### DISCUSSION TITLE
Using YASA for Sleep Staging with Muse-S Headband Data Collected via MindMonitor

### DISCUSSION THREAD

**Discussion by mynameisjustalex** [raphaelvallat/yasa#181] [Q&A] [answered]
Hi YASA Community,

I am working on a research project that involves analyzing EEG data collected from the [Muse-S headband](https://choosemuse.com/pages/muse-s) using the [MindMonitor app](https://mind-monitor.com/Technical_Manual.php#help_streaming), and my team is interested in leveraging YASA for sleep staging and sleep spindle detection with this data.

For context, the Muse-S is a consumer-grade EEG device with four sensors (AF7, AF8, TP9, TP10) placed on the frontal and temporal regions of the head, and the data is recorded at a sampling rate of 256 Hz. The MindMonitor app allows me to export the raw EEG data for further analysis which will my team will try to convert into an EDF/XDF format.

Given the device’s configuration, I wanted to ask for some advice on how best to apply YASA’s sleep staging classifier in this context. 

Specifically, I have the following questions:

1. Single-channel EEG support: Since YASA supports single-channel EEG sleep staging, would it be advisable to focus on the AF7 or AF8 frontal electrodes for sleep staging? Are there any specific preprocessing steps or adjustments you would recommend to optimize the performance of the classifier with Muse-S data?

2. Sleep spindle detection: I am also interested in detecting sleep spindles using the frontal electrodes. Based on your experience, do you think YASA’s spindle detection algorithm will provide reliable results with Muse-S data, or would it be necessary to re-train the spindle detection model?

Thanks in advance for any advice and suggestions! I really appreciate the work that has been done to make YASA and its resources available to the community.

**Comment by raphaelvallat** [comment, marked as answer]
Hi,

I unfortunately have never used the Muse headband or MindMonitor app, and am not familiar with the signal quality of this device. Therefore, please read the answers below with a grain of salt:

1. I would perhaps apply the sleep staging algorithm on both AF7 and AF8 separately — hopefully the predicted sleep stages should be highly consistent. I cannot think of any particular preprocessing step, but I recommend doing some manual checks for the most common artifacts (eye blinks, sweat, disconnected electrodes, etc). My main worry is that the sleep staging algorithm expects the PSG data to have typical PSG amplitudes (with a _mastoid_ reference), and it's possible that the Muse data will be either too small or too large, which will then lead to inaccurate sleep stages.

2. I think that the spindles detection model is somewhat less sensitive than the sleep staging algorithm to "abnormal" data, and therefore I expect that it should work relatively well even on the Muse data. Of course, to be sure you should manually check that the detected spindles are physiologically plausible.

Thank you and best of luck with your research,

Thanks
Raphael

**Reply by mynameisjustalex** [reply]
Hi Dr. Vallat,

Thank you again for your helpful response and for taking the time to address our initial questions. We really appreciate your insights — they’ve been very valuable for our project.

As a quick follow-up, we’re now looking to use YASA with the open source software package ["SleepEEGpy"](https://www.researchgate.net/publication/376630006_SleepEEGpy_a_Python-based_package_for_the_preprocessing_analysis_and_visualization_of_sleep_EEG_data) to extract additional sleep-related features from our Muse-S data, including:

- K-complex detection,
- REM-related features (e.g., REM density),
- Spectral power analysis across different frequency bands (e.g., alpha, beta, delta, theta, gamma).

Given your earlier note about potential limitations with Muse-S data, we were wondering if YASA could also reliably perform these specific analyses, or if there are any additional adjustments we should consider when working with this data.

We fully understand that this is a detailed question and appreciate your time and effort in responding to it.

Thank you again for your time and support!

**Reply by raphaelvallat** [reply]
Hi,

1. YASA does not have a dedicated function for K-complex detection (but see [this discussion](https://github.com/raphaelvallat/yasa/discussions/152))
2. If the eye movements are cleary visible and relatively noise-free, then yes I think YASA should work quite well
3. Bandpower analyses should work pretty well since they are the least susceptible to outliers. However, please note that the frequency content of the Muse data may differ from a ground-truth PSG device and therefore the spectral powers will not be directly comparable with PSG.

Hope this helps,

Thanks,
Raphael
